src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -16,150 +16,199 @@
+                        lastPipeline = ToBackend(cmd->pipeline).Get();
+                        PipelineLayout* pipelineLayout = ToBackend(lastPipeline->GetLayout());
+
+                        // TODO
+                        if (lastPipeline->IsCompute()) {
+                        } else {
+                            commandList->SetGraphicsRootSignature(pipelineLayout->GetRootSignature().Get());
+                            commandList->SetPipelineState(lastPipeline->GetRenderPipelineState().Get());
+                        }
+                    }
+                    break;
+
+                case Command::SetPushConstants:
+                    {
+                        SetPushConstantsCmd* cmd = commands.NextCommand<SetPushConstantsCmd>();
+                    }
+                    break;
+
+                case Command::SetStencilReference:
                     {
                         SetStencilReferenceCmd* cmd = commands.NextCommand<SetStencilReferenceCmd>();
                     }
                     break;
 
-                  case Command::SetBindGroup:
-                      {
-                          SetBindGroupCmd* cmd = commands.NextCommand<SetBindGroupCmd>();
-                      }
-                      break;
-
-                  case Command::SetIndexBuffer:
-                      {
-                          SetIndexBufferCmd* cmd = commands.NextCommand<SetIndexBufferCmd>();
-                      }
-                      break;
-
-                  case Command::SetVertexBuffers:
-                      {
-                          SetVertexBuffersCmd* cmd = commands.NextCommand<SetVertexBuffersCmd>();
-                      }
-                      break;
-
-                  case Command::TransitionBufferUsage:
-                      {
-                          TransitionBufferUsageCmd* cmd = commands.NextCommand<TransitionBufferUsageCmd>();
-                      }
-                      break;
-
-                  case Command::TransitionTextureUsage:
-                      {
-                          TransitionTextureUsageCmd* cmd = commands.NextCommand<TransitionTextureUsageCmd>();
-                      }
-                      break;
-              }
-          }
-      }
+                case Command::SetBindGroup:
+                    {
+                        SetBindGroupCmd* cmd = commands.NextCommand<SetBindGroupCmd>();
+                    }
+                    break;
+
+                case Command::SetIndexBuffer:
+                    {
+                        SetIndexBufferCmd* cmd = commands.NextCommand<SetIndexBufferCmd>();
+
+                        Buffer* buffer = ToBackend(cmd->buffer.Get());
+                        D3D12_INDEX_BUFFER_VIEW bufferView;
+                        bufferView.BufferLocation = buffer->GetD3D12Resource()->GetGPUVirtualAddress() + cmd->offset;
+                        bufferView.SizeInBytes = buffer->GetSize();
```

Comment:
This is the remaining size of the buffer, so it should be ```buffer->GetSize() - cmd.offset```. Also we should add validation that cmd->offset is less than buffer->GetSize() to avoid overflows here (no need to check for the whole size of the draws for now).